### ISSUE TITLE
checking if the value of record.vehicle_id.name is contianing text or not @Aboidrees

### ISSUE THREAD

**Issue by Aboidrees** [odoo/odoo#67796]
Checking if the value of record.vehicle_id.name is containing text or not 
this bug makes a problem when the user select the contract type before selecting the vehicle.

**Here is the error message** 

Odoo Server Error

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 359, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 347, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 531, in response_wrap
    response = f(*args, **kw)
  File "/usr/local/lib/python3.8/dist-packages/odoo/addons/web/controllers/main.py", line 1377, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/addons/web/controllers/main.py", line 1369, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/api.py", line 396, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/api.py", line 383, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/models.py", line 6237, in onchange
    todo = [
  File "/usr/local/lib/python3.8/dist-packages/odoo/models.py", line 6240, in <listcomp>
    if name not in done and snapshot0.has_changed(name)
  File "/usr/local/lib/python3.8/dist-packages/odoo/models.py", line 6071, in has_changed
    return self[name] != record[name]
  File "/usr/local/lib/python3.8/dist-packages/odoo/models.py", line 5669, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/usr/local/lib/python3.8/dist-packages/odoo/fields.py", line 967, in __get__
    self.recompute(record)
  File "/usr/local/lib/python3.8/dist-packages/odoo/fields.py", line 1153, in recompute
    self.compute_value(recs)
  File "/usr/local/lib/python3.8/dist-packages/odoo/fields.py", line 1175, in compute_value
    records._compute_field_value(self)
  File "/usr/local/lib/python3.8/dist-packages/odoo/addons/mail/models/mail_thread.py", line 410, in _compute_field_value
    return super()._compute_field_value(field)
  File "/usr/local/lib/python3.8/dist-packages/odoo/models.py", line 4064, in _compute_field_value
    getattr(self, field.compute)()
  File "/usr/local/lib/python3.8/dist-packages/odoo/addons/fleet/models/fleet_vehicle_cost.py", line 63, in _compute_contract_name
    name = record.cost_subtype_id.name + ' ' + name
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/local/lib/python3.8/dist-packages/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: can only concatenate str (not "bool") to str
```

![image](https://user-images.githubusercontent.com/10165806/111027993-b9fc7e00-8404-11eb-9bed-abc383adee3c.png)

https://github.com/odoo/odoo/blob/24e962b5164917643bfc9be575b5f2417d44ddb3/addons/fleet/models/fleet_vehicle_cost.py#L61-L61

**Comment by Yenthe666** [comment]
Not sure how you managed to get this done because the `name` on that model is required:
https://github.com/odoo/odoo/blob/62a827757168c1044918784b9b6baf8e8a88f4d6/addons/fleet/models/fleet_vehicle.py#L382

**Comment by Aboidrees** [comment]
This problem is happening during the form filling, NOT on the submission. And the code in the bottom execute when we select the name of the vehicle, and also it will execute when we select the contract type.

If I selected the contract type `record.cost_subtype_id.name` before selecting the vehicle `record.vehicle_id.name` this crash will happen. And if I selected the vehicle name first this code it will pass with no problems.

If you noticed the code segment you will find the `record.cost_subtype_id.name` is protected by the `if` statement from being false or empty and the `record.vehicle_id.name` is not 

I am using Arabic RTL. 

```
name = record.vehicle_id.name
if record.cost_subtype_id.name:
     name = record.cost_subtype_id.name + ' ' + name
record.name = name
```
